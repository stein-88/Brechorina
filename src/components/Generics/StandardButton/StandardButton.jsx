import React from 'react'
import sTyle from './StandardButton.scss'

const StandardButton = ({
    text,
    link,
    customClass,
    target,
    onClick,
    isDisabled,
    altern,
    darkBlue,
    customStyle,
}) => {
    if (!text) return null
    const s = customStyle || sTyle
    if (onClick) {
        return (
            <span
                aria-hidden="true"
                onClick={(e) => {
                    if (!isDisabled) {
                        onClick(e)
                    }
                }}
                className={`${s.standardButton} ${darkBlue ? s.darkBlue : ''} ${isDisabled ? s.disabledButton : ''} ${altern ? s.alternative : ''} ${customClass || ''}`}
            >
                {text}
            </span>
        )
    }
    return (
        <a href={link} rel={target ? 'noopener noreferrer' : 'alternate'} target={target ? '_blank' : '_self'} className={`${s.standardButton} ${darkBlue ? s.darkBlue : ''} ${altern ? s.alternative : ''} ${customClass || ''}`}>{text}</a>
    )
}

export default StandardButton
