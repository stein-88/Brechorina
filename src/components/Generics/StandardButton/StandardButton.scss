.standardButton {
    cursor: pointer;
    padding: 10px 4rem;
    color: white !important;
    display: inline-block;
    text-align: center;
    outline: 0;
    background-color: #69afe3;
    border-radius: 25px;
    border-color: transparent;
    text-decoration: none !important;

    &.darkBlue {
        background-color: #0075ce;

        &.disabledButton {
            cursor: default;
            background-color: #cacaca;
            border-color: #cacaca;

            &:hover {
                background-color: #cacaca;
            }
        }

        &:hover {
            background-color: lighten($color: #0075ce, $amount: 10);
            border-color: lighten($color: #0075ce, $amount: 10);
        }
    }

    &.alternative {
        background-color: white;
        border: 1px solid #69afe3;
        color: #69afe3 !important;

        &.disabledButton {
            cursor: default;
            background-color: #cacaca;
            border-color: #cacaca;

            &:hover {
                background-color: #cacaca;
            }
        }

        &.darkBlue {
            color: #0075ce !important;
            border-color: #0075ce;

            &.disabledButton {
                cursor: default;
                background-color: #cacaca;
                border-color: #cacaca;

                &:hover {
                    background-color: #cacaca;
                }
            }

            &:hover {
                color: white;
                background-color: lighten($color: #0075ce, $amount: 10);
                border-color: lighten($color: #0075ce, $amount: 10);
            }
        }

        &:hover {
            background-color: darken($color: #69afe3, $amount: 10);
            color: white !important;
        }
    }

    &.disabledButton {
        cursor: default;
        background-color: #cacaca;
        border-color: #cacaca;

        &:hover {
            background-color: #cacaca;
        }
    }

    &:focus {
        outline: 0;
    }

    &:hover {
        background-color: darken($color: #69afe3, $amount: 10);
    }
}