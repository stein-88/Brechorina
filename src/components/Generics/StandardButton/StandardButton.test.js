import React from 'react'
import renderer from 'react-test-renderer'
import StandardButton from './StandardButton'

describe('Test on <StandardButton /> component', () => {
    test('Should return null without props', () => {
        const componentNoprops = renderer.create(<StandardButton />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toBeNull()
    })

    test('Should render and return a instead of span', () => {
        const componentNoprops = renderer.create(<StandardButton text="Ciapush" link="/ciapek" />)
        const treeNoprops = componentNoprops.toJSON()
        const fejiao = componentNoprops.root.findAllByProps({ href: '/ciapek' })
        expect(fejiao).toHaveLength(1)
        expect(treeNoprops.type).toEqual('a')
        expect(treeNoprops).toMatchSnapshot()
    })

    test('Should render and return span instead of a', () => {
        const componentNoprops = renderer.create(<StandardButton text="Ciapush" customClass="randomClass" onClick={() => null} />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops.type).toEqual('span')
        expect(treeNoprops.props.className).toEqual(expect.stringContaining('randomClass'))
        expect(treeNoprops.children).toHaveLength(1)
        expect(treeNoprops.children[0]).toEqual('Ciapush')
        expect(treeNoprops).toMatchSnapshot()
    })
})
