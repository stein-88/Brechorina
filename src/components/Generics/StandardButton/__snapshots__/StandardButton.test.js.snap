// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <StandardButton /> component Should render and return a instead of span 1`] = `
<a
  className="standardButton   "
  href="/ciapek"
  rel="alternate"
  target="_self"
>
  Ciapush
</a>
`;

exports[`Test on <StandardButton /> component Should render and return span instead of a 1`] = `
<span
  aria-hidden="true"
  className="standardButton    randomClass"
  onClick={[Function]}
>
  Ciapush
</span>
`;
